### PR TITLE
Add tests for GFp

### PR DESCRIPTION
**Pull request by armfazh**

### CODE REVIEW

File: gfp_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
+package bn256
```
Comment (cjpatton) [inline gfp_test.go:1]: Should files have CF copyright on them?

File: gfp_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"io"
+	"math/big"
+	"testing"
+)
+
+func randomGF(r io.Reader) *big.Int {
+	k, err := rand.Int(r, p)
```
Comment (cjpatton) [inline gfp_test.go:13]: Where is `p` defined?

Comment (armfazh) [reply gfp_test.go:13]: in the constants.go file
https://github.com/cloudflare/bn256/blob/master/constants.go#L16

File: gfp_test.go (hunk 1 above)
Comment (armfazh) [reply gfp_test.go:1]: the main license file is from the go original project, and I assume that applies to all the files in the repo.

**Review by cjpatton — APPROVED** [review]
Seems reasonable. Comments describing how the helper functions work (e.g., `togfP()`) would be really helpful for readability.

**Comment by armfazh** [comment]
> Seems reasonable. Comments describing how the helper functions work (e.g., `togfP()`) would be really helpful for readability.

I added some.

**Pull request merged**